docs/src/further_topics/ugrid/data_model.rst

```diff
@@ -0,0 +1,319 @@
+.. include:: ../../common_links.inc
+
+.. _ugrid model:
+
+The Mesh Data Model
+*******************
+
+.. important::
+
+        This page is intended to summarise the essentials that Iris users need
+        to know about meshes. For exhaustive details on UGRID itself:
+        `visit the official UGRID conventions site`__.
+
+Evolution, not revolution
+=========================
+Mesh support has been designed wherever possible to fit within the existing
+Iris model. Meshes concern only the spatial location of data, and can
+optionally be limited to just the horizontal location (e.g. X and Y). Other
+dimensions such as time or experimental re-runs (and often vertical levels)
+retain their familiar structured format.
+
+The UGRID conventions themselves are designed as an addition to the existing CF
+conventions, which are at the core of Iris' philosophy.
+
+What's Different?
+=================
+
+The mesh format represents data's geographic location using an **unstructured
+mesh**. This has significant pros and cons when compared to a structured grid.
+
+.. contents::
+   :local:
+
+The Detail
+----------
+..
+    The diagram images are SVG's, so editable by any graphical software
+     (e.g. Inkscape). They were originally made in MS PowerPoint.
+
+    Uses the IBM Colour Blind Palette (see
+     http://ibm-design-language.eu-de.mybluemix.net/design/language/resources/color-library
+     )
+
+Structured Grids
+~~~~~~~~~~~~~~~~
+Assigning data to locations using a structured grid is essentially an act of
+matching coordinate arrays to each dimension of the data array. The data can
+also be represented as an area (instead of a point) by including a bounds array
+for each coordinate array.
+
+.. figure:: images/data_structured_grid.svg
+   :alt: Diagram of how data is represented on a structured grid
+
+   Data on a structured grid
```

Comment:
Are there references in the text pointing to these images? I'm not sure I fully follow what the circles and crosses mean, e.g. are there x-y axes on the cross diagrams, but index axes on the dot diagrams? And the diamonds are scalar values mapped to those coords?

![image](https://user-images.githubusercontent.com/22805/150578762-a36d6b0e-4b9d-48f9-bf45-8092dd5b11de.png)

Reply:
I've been trying to keep the prose and diagrams self-contained - for accessibility and ease of maintenance.

And I had also hoped that the diagrams would be self-describing where possible. But from yours and others' feedback there are parts that aren't as obvious as I had thought! I'll see what I can think of to make things clearer within the diagrams, while avoiding information overload.

Reply:
>But from yours and others' feedback there are parts that aren't as obvious as I had thought! I'll see what I can think of to make things clearer within the diagrams, while avoiding information overload.

Other opinions on this would be gratefully received @SciTools/developers

Reply:
Is the vague highlighting of the top-left point intentional?
I think I'd prefer it wasn't highlighted, and just one in the middle was? I'd also add some form of background colouring so that it's clearer what's the data, what's the objects that are part of the cube and what's just conceptual process.

Reply:
Ah, I see now that it was. Can we make the two highlight types more distinct?

Reply:
I've had a go at addressing both:

* The distinction between arrays and geography - by including an Africa outline to indicate geography.
* The highlighting of two distinct points - bolder dash markings for the 'secondary' point.

3c0e5f85bff8e008226dc9d8a88da47b9b6f4db1

Reply:
Considering there is a lot going on in this, I think it would be quite difficult for a user to follow this diagram without a bit of prose explaining
 what is going on. Maybe include something like the following in the figure caption:

> 1 dimensional Coordinate Arrays, x and y (pink circles)  are combined to construct a structured grid of points (pink crosses). If bounds arrays (blue circles) are present, the x and y bounds construct 4-sided cells (outlined with blue). The data array (orange circles) assigns data with the points (orange diamonds) if no bounds are present) or with the bounded cells (orange quadrialterals) when bounds exist. The above example highlighs the location (x=25, y=20) which has bounds (x=[23,28], y=[19,21]) 

I would also think the other figures would benefit from a more detailed caption

Reply:
> I've been trying to keep the prose and diagrams self-contained - for accessibility and ease of maintenance.
> 
> And I had also hoped that the diagrams would be self-describing where possible. But from yours and others' feedback there are parts that aren't as obvious as I had thought! I'll see what I can think of to make things clearer within the diagrams, while avoiding information overload.

I understand that rationale, makes a lot of sense so feel free to ignore this advice if you feel those concerns trump this.

I know this isn't a paper or publication, but I was taught (and I think it is a good general rule of thumb) that all figures in a document should be referenced from the text, otherwise what is the context, why are they there?
A brief search for a guide to this I found the APA style guide, see https://guides.lib.monash.edu/citing-referencing/apa-tables-figures:

> * All Tables and Figures must be referred to in the main body of the text.

Reply:
We're gonna go with some more explicit captions on each figure, and I'll think about some cross-referencing with the text blocks.

Reply:
3692c1cfa51f15181437b22088f3f223d0037a93

Reply:
Also I hadn't spotted that the example coordinates were unbelievable in a couple of ways.

67b1eb3765829eb70b5d4f2e581a8b8dde09b498